config.c

```diff
@@ -2515,6 +2515,12 @@ int git_config_get_max_percent_split_change(void)
 
 int repo_config_get_fsmonitor(struct repository *r)
 {
+	if (!r->worktree) {
+		/* FSMonitor makes no sense in bare repositories */
+		core_fsmonitor = 0;
```

Comment:
This should be `= NULL`, otherwise, sparse will complain about "Using plain integer as NULL pointer".